### PR TITLE
Refactor: Rename Orchestrator to VTOrc in the codebase

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR renames the occurrences of `orchestrator` with `vtorc` in package names, function names, comments, logs, etc.
Some usages of `orchestrator` are still left since they actually refer to the original orchestrator.

VTop changes are also required for this PR  since the web directory has also changed from `web/orchestrator` to `web/vtorc`

Configuration parameters having `Orchestrator` in their name have not been refactored yet.

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by deepthi** [comment]
If someone is already running VTOrc with vitess-operator, what is the migration path for them?

**Comment by GuptaManan100** [comment]
There isn't. When they upgrade their operator version, they'll have to update the VTOrc image too.

**Comment by GuptaManan100** [comment]
One way to make it work is to copy the `web/orchestrator` directory into `web/vtorc` but not remove the first. Then we can remove it after v15 is released.

**Comment by deepthi** [comment]
> There isn't. When they upgrade their operator version, they'll have to update the VTOrc image too.

Can we just document this in the `Deployment Notes` and release notes for both Vitess 15 and VTop 2.8 releases? That vitess and vitess-operator will need to be upgraded together?
The usual practice is to upgrade the operator and then rollout the vitess version change. Will that work?

File: go/vt/vtorc/db/tls.go [hunk 1]
```diff
@@ -124,15 +124,15 @@ func SetupMySQLTopologyTLS(uri string) (string, error) {
 }
 
 // Create a TLS configuration from the config supplied CA, Certificate, and Private key.
-// Register the TLS config with the mysql drivers as the "orchestrator" config
+// Register the TLS config with the mysql drivers as the "vtorc" config
 // Modify the supplied URI to call the TLS config
-func SetupMySQLOrchestratorTLS(uri string) (string, error) {
-	if !orchestratorTLSConfigured {
+func SetupMySQLVTOrcTLS(uri string) (string, error) {
+	if !vtorcTLSConfigured {
```
Comment (rsajwani) [inline go/vt/vtorc/db/tls.go:130]: I am observing all the flags and function name are VTOrc then why is vtorcTLSConfigured not following this convention?

File: go/vt/vtorc/db/tls.go [hunk 2]
```diff
@@ -124,15 +124,15 @@ func SetupMySQLTopologyTLS(uri string) (string, error) {
...
 		tlsConfig, err := ssl.NewTLSConfig(config.Config.MySQLOrchestratorSSLCAFile, !config.Config.MySQLOrchestratorSSLSkipVerify)
 		// Drop to TLS 1.0 for talking to MySQL
 		tlsConfig.MinVersion = tls.VersionTLS10
 		if err != nil {
-			log.Fatalf("Can't create TLS configuration for Orchestrator connection %s: %s", uri, err)
+			log.Fatalf("Can't create TLS configuration for VTOrc connection %s: %s", uri, err)
```
Comment (rsajwani) [inline go/vt/vtorc/db/tls.go:135]: VTOrc vs Vtorc

File: go/vt/vtorc/inst/instance_topology.go [hunk 3]
```diff
@@ -2070,7 +2070,7 @@ func relocateReplicasInternal(replicas []*Instance, instance, other *Instance) (
 }
 
 // RelocateReplicas will attempt moving replicas of an instance indicated by instanceKey below another instance.
-// Orchestrator will try and figure out the best way to relocate the servers. This could span normal
+// VTOrc will try and figure out the best way to relocate the servers. This could span normal
```
Comment (rsajwani) [inline go/vt/vtorc/inst/instance_topology.go:2073]: VTOrc vs Vtorc

File: go/vt/vtorc/inst/tablet_dao.go [hunk 4]
```diff
@@ -45,7 +45,7 @@ var ErrTabletAliasNil = errors.New("tablet alias is nil")
 // SwitchPrimary makes the new tablet the primary and proactively performs
 // the necessary propagation to the old primary. The propagation is best
 // effort. If it fails, the tablet's shard sync will eventually converge.
-// The proactive propagation allows a competing Orchestrator from discovering
+// The proactive propagation allows a competing VTOrc from discovering
```
Comment (rsajwani) [inline go/vt/vtorc/inst/tablet_dao.go:48]: VTOrc vs Vtorc

**Review by rsajwani — APPROVED** [review]
LGTM. Overall I am observing you are using VTOrc for functions names, variable names and flags name. But using Vtorc in comments and log statements. I did find some inconsistency with the convention so have pointed out in the review.  Feel free to correct me if my observation is not correct.

**Comment by rsajwani** [comment]
There are some static code analysis check failures. please fix them before you  push the code

File: go/vt/vtorc/app/cli.go [hunk 5]
```diff
@@ -1116,7 +1116,7 @@ func Cli(command string, strict bool, instance string, destination string, owner
 			fmt.Println(instanceKey.DisplayString())
 		}
 		// Recovery & analysis
-	case registerCliCommand("recover", "Recovery", `Do auto-recovery given a dead instance`), registerCliCommand("recover-lite", "Recovery", `Do auto-recovery given a dead instance. Orchestrator chooses the best course of actionwithout executing external processes`):
+	case registerCliCommand("recover", "Recovery", `Do auto-recovery given a dead instance`), registerCliCommand("recover-lite", "Recovery", `Do auto-recovery given a dead instance. VTOrc chooses the best course of actionwithout executing external processes`):
```
[the remark this answers is not in the document]
Comment (GuptaManan100) [reply go/vt/vtorc/app/cli.go:1119]: VTOrc in the capitalized form and vtorc in the lower cased version

File: go/test/endtoend/vtorc/utils/utils.go [hunk 6]
```diff
@@ -269,7 +269,7 @@ func StopVtorcs(t *testing.T, clusterInfo *VtOrcClusterInfo) {
 	clusterInfo.ClusterInstance.VtorcProcesses = nil
 }
 
-// SetupVttabletsAndVtorc is used to setup the vttablets and start the orchestrator
+// SetupVttabletsAndVtorc is used to setup the vttablets and start the vtorcs
```
[the remark this answers is not in the document]
Comment (GuptaManan100) [reply go/test/endtoend/vtorc/utils/utils.go]: Fixed function name

**Comment by GuptaManan100** [comment]
@rsajwani We want to use `VTOrc` for the capitalized form and `vtorc` when lowercased. I have gone ahead and changed all `Vtorc` to `VTOrc` too in the codebase. This should make everything consistent.

**Comment by GuptaManan100** [comment]
I have made the changes like we discussed @deepthi. I have added a symlink from `web/orchestrator` to `web/vtorc` and changed the dockerfiles to include both. So, the changes are now non-breaking. I have tested the new built image with vitess-operator and it works without any changes. I have also checked the compose example and local examples, to verify that they continue to work.

**Review by deepthi — APPROVED** [review]
LGTM.
> I have tested the new built image with vitess-operator and it works without any changes. I have also checked the compose example and local examples, to verify that they continue to work.

Thank you for the diligence in testing

**Comment by rsajwani** [comment]
> @rsajwani We want to use `VTOrc` for the capitalized form and `vtorc` when lowercased. I have gone ahead and changed all `Vtorc` to `VTOrc` too in the codebase. This should make everything consistent.

Ah make sense. Thanks

**Review by rsajwani — APPROVED** [review]

**Comment by matthiasr** [comment]
Or will the operator understand the difference and handle it automatically?

**Pull request merged**